docs/user/quickstart.rst

```diff
@@ -353,4 +398,234 @@ This would print for this example application:
+
+
+    class JSONTranslator:
+        # NOTE: Normally you would simply use req.get_media() and resp.media for
+        # this particular use case; this example serves only to illustrate
+        # what is possible.
+
+        async def process_request(self, req, resp):
+            # req.stream corresponds to the WSGI wsgi.input environ variable,
+            # and allows you to read bytes from the request body.
+            #
+            # See also: PEP 3333
+            if req.content_length in (None, 0):
+                # Nothing to do
+                return
+
+            body = await req.stream.read()
+            if not body:
+                raise falcon.HTTPBadRequest(title='Empty request body',
+                                            description='A valid JSON document is required.')
+
+            try:
+                req.context.doc = json.loads(body.decode('utf-8'))
+
+            except (ValueError, UnicodeDecodeError):
+                raise falcon.HTTPError(falcon.HTTP_753,
+                                       title='Malformed JSON',
+                                       description='Could not decode the request body. The '
+                                       'JSON was incorrect or not encoded as '
+                                       'UTF-8.')
+
+        async def process_response(self, req, resp, resource, req_succeeded):
+            if not hasattr(resp.context, 'result'):
+                return
+
+            resp.body = json.dumps(resp.context.result)
+
+
+    def max_body(limit):
+
+        async def hook(req, resp, resource, params):
+            length = req.content_length
+            if length is not None and length > limit:
+                msg = ('The size of the request is too large. The body must not '
+                       'exceed ' + str(limit) + ' bytes in length.')
+
+                raise falcon.HTTPPayloadTooLarge(
+                    title='Request body is too large', description=msg)
+
+        return hook
+
+
+    class ThingsResource:
+
+        def __init__(self, db):
+            self.db = db
+            self.logger = logging.getLogger('thingsapp.' + __name__)
+
+        async def on_get(self, req, resp, user_id):
+            marker = req.get_param('marker') or ''
+            limit = req.get_param_as_int('limit') or 50
+
+            try:
+                result = await self.db.get_things(marker, limit)
+            except Exception as ex:
+                self.logger.error(ex)
+
+                description = ('Aliens have attacked our base! We will '
+                               'be back as soon as we fight them off. '
+                               'We appreciate your patience.')
+
+                raise falcon.HTTPServiceUnavailable(
+                    title='Service Outage',
+                    description=description,
+                    retry_after=30)
+
+            # NOTE: Normally you would use resp.media for this sort of thing;
```

Comment:
Better to stay consistent with the previous mention of `resp.get_media()` IMHO, otherwise we might need to add a sentence or two about awaiting properties.

Reply:
This should be fine, right? You may have been thinking about the request, not the response... 😄

Reply:
Heh, true :laughing: Not sure what I was thinking here :slightly_smiling_face: